File: apps/desktop/frontend/src/types/DetailContent.ts
import type { DetailImage, IconReference } from "./index";

export type DetailContent =
    | { alternative_text: string; kind: "image"; source: DetailImage; }
    | { files: Array<{ icon: IconReference | null; name: string; path: string; }>; kind: "files"; }
    | { kind: "text"; value: string; };

File: engine/extension-protocol/src/DetailContent.rs
use serde::{Deserialize, Serialize};

use crate::ImageSource;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(tag = "kind", rename_all = "camelCase")]
pub enum DetailContent {
    Text {
        value: String,
    },
    Files {
        files: Vec<crate::ViewFile>,
    },
    Image {
        source: ImageSource,
        alternative_text: String,
    },
}
